src/main/java/hudson/plugins/collabnet/util/Helper.java

```diff
@@ -0,0 +1,78 @@
+package hudson.plugins.collabnet.util;
+
+import hudson.model.AbstractBuild;
+import hudson.model.Result;
+import hudson.model.Run;
+import net.sf.json.JSONObject;
+import org.apache.http.HttpResponse;
+import org.apache.http.NameValuePair;
+import org.apache.http.client.HttpClient;
+import org.apache.http.client.entity.UrlEncodedFormEntity;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.impl.client.DefaultHttpClient;
+import org.apache.http.message.BasicNameValuePair;
+import org.apache.http.util.EntityUtils;
+
+import java.io.IOException;
+import java.io.PrintStream;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.logging.Logger;
+
+public class Helper {
+
+    /** Prefix for messages appearing in the console log, for readability */
+    private static String LOG_MESSAGE_PREFIX = "TeamForge WEBR Build Notifier - ";
+
+    static Logger logger = Logger.getLogger("hudson.plugins.collab.orchestrate");
+
+    public static String getToken(URL ctfUrl, String ctfUserName, String ctfPassword) throws IOException {
+        String end_point = ctfUrl.toString()+"/oauth/auth/token";
+        HttpClient httpclient = new DefaultHttpClient();
```

Comment:
Please make sure HttpClient/Connections are closed/disposed properly. Also, CollabNet plugin allows ignoring SSL warnings/errors, however this implementation does not take the setting into account. See `CNFormFieldValidator.java` for an example. This should apply to all HTTP connections.

Reply:
Sure, will update this.